src/learner-record/model/courseRecord/courseRecord.ts

```diff
@@ -0,0 +1,83 @@
+import { plainToClass } from 'class-transformer'
+import { ModuleRecord } from "../moduleRecord/moduleRecord"
+import { Record, RecordState } from "../record"
+
+
+export enum CourseRecordPreference {
+	Liked = 'LIKED',
+	Disliked = 'DISLIKED'
+}
+
+export class CourseRecordResponse {
+	courseRecords: CourseRecord[]
+
+	constructor(courseRecords: CourseRecord[]) {
+		this.courseRecords = courseRecords
+	}
+}
+
+export class CourseRecord extends Record {
+	courseTitle: string
+	modules: ModuleRecord[]
+	preference?: CourseRecordPreference
+	lastUpdated?: Date
+	courseDisplayState?: string
+	required: boolean
+
+	constructor(
+		courseId: string,
+		userId: string,
+		state: RecordState = RecordState.Null,
+		modules: ModuleRecord[] = [],
+		courseTitle: string,
+		required: boolean,
+		preference?: CourseRecordPreference,
+		lastUpdated?: Date
+	) {
+		super(courseId, userId, state)
+		this.courseTitle = courseTitle
+		this.preference = preference
+		this.required = required
+		if (lastUpdated) {
+			this.lastUpdated = new Date(lastUpdated)
+		}
+
+		if (modules.length > 0) {
+			this.fillRecords(modules)
+		}
+	}
+
+	public updateModuleRecord(moduleRecordId: number, moduleRecord: ModuleRecord) {
+		for (let i = 0; i < this.modules.length; i++) {
+			const mr = this.modules[i];
+			if (mr.id === moduleRecordId) {
+				this.modules[i] = moduleRecord
+			}
+		}
+	}
```

Comment:
You can use the array `findIndex` function here for a much easier and more efficient way to to update a module.

Example:
```ts
let index = this.modules.findIndex(module => module.id === moduleRecordId)
this.modules[index] = moduleRecord
```

Reply:
I've pushed a change to rectify this, however it doesn't appear to be coming up in the PR...

Reply:
Hmm... I don't see it. Try pushing again maybe?

Reply:
Very weird https://github.com/Civil-Service-Human-Resources/lpg-management/blob/60e3110504dc3d0ccca6afbe4da1143f8a20b9f6/src/learner-record/model/courseRecord/courseRecord.ts#L51

Just pushed again, still not having it

Reply:
I can see the updated file in the "Files changed" tab